### PR TITLE
[SYCL] add dependent-load flag to exclude CWD from default search path

### PR DESCRIPTION
**Pull request by ykhatav**

### CODE REVIEW

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 1]
```diff
@@ -99,6 +99,10 @@ add_sycl_plugin(level_zero
     ${XPTI_LIBS}
 )
 
+if (WIN32)
+ target_link_options(pi_level_zero PUBLIC /DEPENDENTLOADFLAG:0x800)
```
Comment (kraszkow) [inline sycl/plugins/level_zero/CMakeLists.txt:103]: I would suggest adding comment what "magic number" 0x800 is (LOAD_LIBRARY_SEARCH_SYSTEM32).

**Review by kraszkow — CHANGES_REQUESTED** [review]

**Review by againull — APPROVED** [review]

**Review by kraszkow — APPROVED** [review]
LGTM

**Comment by againull** [comment]
@unified-runtime-reviewers Could you please review.

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 2]
```diff
@@ -99,6 +99,11 @@ add_sycl_plugin(level_zero
     ${XPTI_LIBS}
 )
 
+if (WIN32)
+ # 0x800: Search for the DLL only in the System32 folder
+ target_link_options(pi_level_zero PUBLIC /DEPENDENTLOADFLAG:0x800)
```
Comment (aarongreig) [inline sycl/plugins/level_zero/CMakeLists.txt:104]: this file uses two spaces for indentation
```suggestion
  # 0x800: Search for the DLL only in the System32 folder
  target_link_options(pi_level_zero PUBLIC /DEPENDENTLOADFLAG:0x800)
```

**Review by aarongreig — CHANGES_REQUESTED** [review]

**Pull request merged**